### PR TITLE
Adding option to display similar names to the package entered

### PR DESCRIPTION
**Pull request by Robbie-Cook**
See #22 .

Now, the usage can be `npm-name hello --similar`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,12 +44,19 @@
...
+		"thesaurus": "0.0.1"
 	},
 	"devDependencies": {
 		"ava": "^2.1.0",
 		"execa": "^4.0.0",
 		"xo": "^0.26.1"
+	},
+	"xo": {
+		"rules": {
+			"no-await-in-loop": [0]
```
Comment (sindresorhus) [inline package.json:59]: It's better to use an inline `// eslint-disable` comment for this.

File: util.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ * @param {{
+ *   name: string;
+ *   isAvailable: boolean;
+ *   isOrganization: boolean;
+ * 	 isSquatter?: boolean;
+ *  }} pkg
+ * @return {Promise<Array<string> | void>}
+ */
+async function getSimilarPackageNames(pkg) {
+	if (pkg.name) {
```
Comment (sindresorhus) [inline util.js:52]: Use an early-return.

File: util.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+		const name = pkg.name.replace(/@/, '');
+		const names = thesaurus.find(name);
+		if (names) {
```
Comment (sindresorhus) [inline util.js:55]: Use an early-return.

File: util.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+			return ret;
+		})
+	);
+
+	return packages;
+}
+
+/**
+ * Gets a list of similar package names that are available.
+ *
```
Comment (sindresorhus) [inline util.js:42]: I prefer not to use the `*` border.

Comment (sindresorhus) [reply util.js:42]: https://github.com/sindresorhus/typescript-definition-style-guide

File: package.json [hunk 5]
```diff
@@ -44,12 +44,19 @@
 		"npm-name": "^6.0.0",
 		"ora": "^4.0.3",
 		"org-regex": "^1.0.0",
+		"slugify": "^1.4.5",
```
Comment (sindresorhus) [inline package.json:47]: Use https://github.com/sindresorhus/slugify

File: cli.js [hunk 6]
```diff
@@ -53,28 +68,59 @@ function log(pkg) {
...
+				}
+
+				secondSpinner.stop();
+			}
+		}
 
-	for (const pkg of packages) {
-		log(pkg);
+		process.exit(
+			packages.every(pkg => Boolean(pkg.isAvailable || pkg.isSquatter)) ? 0	: 2
```
Comment (sindresorhus) [inline cli.js:119]:
```suggestion
			packages.every(pkg => Boolean(pkg.isAvailable || pkg.isSquatter)) ? 0 : 2
```

File: util.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
+const squatter = require('squatter');
```
Comment (sindresorhus) [inline util.js:1]: You need to add this file to `files` in package.json.

File: cli.js [hunk 8]
```diff
@@ -3,21 +3,27 @@
...
 const ora = require('ora');
-const organizationRegex = require('org-regex')({exact: true});
+const {getSimilarPackageNames, checkNames} = require('./util');
 
-const cli = meow(`
+const cli = meow(
+	`
 	Usage
-	  $ npm-name <name> …
+		$ npm-name <name> …
```
Comment (sindresorhus) [inline cli.js:13]: Don't do unrelated changes.

File: cli.js [hunk 9]
```diff
@@ -3,21 +3,27 @@
...
 
 	Examples
 	  $ npm-name chalk
 	  ${logSymbols.error} ${chalk.bold('chalk')} is unavailable
 	  $ npm-name abc123
 	  ${logSymbols.warning} ${chalk.bold('abc123')} is squatted
+	  $ npm-name --similar bigiron
+		${logSymbols.warning} ${chalk.bold('bigiron')} is unavailable
+		\nSimilar names:\n
+		${logSymbols.success} ${chalk.bold('bigcactus')} is available;
```
Comment (sindresorhus) [inline cli.js:26]: Incorrect indentation.

File: cli.js [hunk 10]
```diff
@@ -3,21 +3,27 @@
...
+	Options
+		--similar  Find similar package names
 
 	Examples
 	  $ npm-name chalk
 	  ${logSymbols.error} ${chalk.bold('chalk')} is unavailable
 	  $ npm-name abc123
 	  ${logSymbols.warning} ${chalk.bold('abc123')} is squatted
+	  $ npm-name --similar bigiron
+		${logSymbols.warning} ${chalk.bold('bigiron')} is unavailable
```
Comment (sindresorhus) [inline cli.js:24]: You also need to update the readme output.

File: cli.js [hunk 11]
```diff
@@ -53,28 +68,59 @@ function log(pkg) {
 	}
 }
 
-const spinner = ora(`Checking ${input.length === 1 ? 'name' : 'names'} on npmjs.com…`).start();
+// Main
+
+const spinner = ora(
+	`Checking ${input.length === 1 ? 'name' : 'names'} on npmjs.com…`
+).start();
```
Comment (sindresorhus) [inline cli.js:75]: Don't do unrelated changes.

File: cli.js [hunk 12]
```diff
@@ -53,28 +68,59 @@ function log(pkg) {
...
 
 (async () => {
-	const result = await npmName.many(input);
+	const packages = await checkNames(input);
+	spinner.stop();
 
-	const packages = await Promise.all([...result].map(async ([name, isAvailable]) => {
-		const ret = {name, isAvailable, isOrganization: organizationRegex.test(name)};
+	if (packages) {
```
Comment (sindresorhus) [inline cli.js:81]: Invert this condition to reduce nesting.

File: cli.js [hunk 13]
```diff
@@ -53,28 +68,59 @@ function log(pkg) {
...
+		for (const pkg of packages) {
```
Comment (sindresorhus) [inline cli.js:82]:
```suggestion
		for (const package_ of packages) {
```

File: cli.js [hunk 14]
```diff
@@ -3,21 +3,27 @@
...
 
-const cli = meow(`
+const cli = meow(
+	`
 	Usage
-	  $ npm-name <name> …
+		$ npm-name <name> …
+
+	Options
+		--similar  Find similar package names
```
Comment (sindresorhus) [inline cli.js:16]:
```suggestion
		--similar  Find similar package names too
```

**Comment by sindresorhus** [comment]
Can you also update the FAQ section in the readme? https://github.com/sindresorhus/npm-name-cli#why-would-i-use-npm-name-rather-than-npms-built-in-search

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by Robbie-Cook** [comment]
Sorry, but I'm not comfortable working on this PR. I am not going to make those changes you requested. Feel free to close this.
That being said, please use this PR how you like, anyone. Would love to see a fixed version of this merged in.

**Comment by sindresorhus** [comment]
> Sorry, but I'm not comfortable working on this PR.

Any reason why?

**Pull request closed without merging**